# Project GeoSpatial

## Introduction

This project consist in selecting the optimal location for a gaming enterprise conditioned by some nearby locations:

- 20 Designers like to go to design talks and share knowledge. There must be some nearby companies that also do design.
- 30% of the company staff have at least 1 child.
- 15 Developers like to be near successful tech startups that have raised at least 1 Million dollars.
- 10 Executives like Starbucks A LOT. Ensure there's a starbucks not too far.
- 20 Account managers need to travel a lot.
- Everyone in the company is between 25 and 40, give them some place to go party.
- 1 The CEO is vegan.
- 1 If you want to make the maintenance guy happy, a basketball stadium must be around 10 Km.
- The office dog—"Dobby" needs a hairdresser every month. Ensure there's one not too far away.


## Libraries

First of all, we install/import all the necessary libraries:

In [1]:
%pip install pyjsonviewer
!pip install cartoframes
%pip install geopandas

Note: you may need to restart the kernel to use updated packages.
Note: you may need to restart the kernel to use updated packages.


In [3]:
import requests
import json
from dotenv import load_dotenv
import os
import pandas as pd
import sys
sys.path.append('/Users/juanperezdeayala/Documents/IRONHACK/Geospatial-Data-Project-/SRC')
from functools import reduce
import src.datagenerator as dg
import src.index as ix
import operator
import geopandas as gpd
import shapely.geometry
from pymongo import MongoClient
from pymongo import GEOSPHERE

## Selecting locations


To select the location, I have chosen three top companies in the videogames market and I will select which of them covers better the conditions.

#### Top 8 Companies in the sector

- Tencent: Shenzen, China
- Sony Interactive Entertainment: Tokio, Japón
- Xbox Game Studios: Washington, USA
- Activision Blizzard: Los Ángeles, California, USA
- Electronic Arts: Redwood city, California, USA
- Ubisoft: Montreuil, Francia
- Square Enix: Tokio, Japón
- Nintendo: Kioto, Japón

I have chosen Sony Interactive Entertainment, Activision Blizzard and Ubisoft in order to select and compare different countries.

## Getting Coordinates

In order to get the coordinates (lat y long), I have created a function where you input the adress and it output the coordinates.

Using the function with Sony

In [5]:
Tokio_coord = dg.geocode('Minato City, Tokyo 108-0075, Japon')
Tokio_coord

{'standard': {'addresst': 'Minato',
  'city': 'Tokyo',
  'prov': 'JP',
  'countryname': 'Japan',
  'postal': {},
  'confidence': '0.4'},
 'longt': '139.74342',
 'alt': {},
 'elevation': {},
 'latt': '35.65278'}

Using the function with Activision

In [6]:
USA_coord = dg.geocode('3100 Ocean Park Blvd, Santa Monica')
USA_coord

{'standard': {'stnumber': '3100',
  'addresst': 'Ocean Park Blvd',
  'postal': '90405',
  'region': 'CA',
  'zip': '90405',
  'prov': 'US',
  'city': 'Santa Monica',
  'countryname': 'United States of America',
  'confidence': '1'},
 'longt': '-118.45255',
 'alt': {},
 'elevation': {},
 'latt': '34.01978'}

Using the function with Ubisoft

In [7]:
France_coord = dg.geocode('126 Rue de Lagny, 93100 Montreuil')
France_coord

{'standard': {'stnumber': '126',
  'addresst': 'Rue De Lagny',
  'city': 'Montreuil',
  'prov': 'FR',
  'countryname': 'France',
  'postal': '93100',
  'confidence': '0.9'},
 'longt': '2.41788',
 'alt': {},
 'elevation': {},
 'latt': '48.84938'}

## Getting the Score for each location

To achieve the score for each country, I have selected some of the initial conditions since it is impossible to meet all the requirements:

- 30% of the company staff have at least 1 child: SCHOOLS
- Executives like Starbucks A LOT: STARBUCKS
- Account managers need to travel a lot: TRAIN STATIONS / AIRPORTS
- Everyone in the company is between 25 and 40, give them some place to go party: CLUBS

In addition, to obtain a scoring system, I will perform weightings for each element:
- SCHOOLS: 50
- STARBUCKS: 5
- TRAIN STATION: 10
- AIRPORT: 10
- CLUBS: 25

#### Score Tokyo

In [8]:
ix.main('Minato City, Tokyo 108-0075, Japon')

City: Tokyo, Score:1645, location: 35.65278,139.74342, Schools: 20, Starbucks: 7, Night Clubs: 20, Train Stations: 8 Airports: 3


#### Score Santa Monica

In [10]:
ix.main('3100 Ocean Park Blvd, Santa Monica')

City: Santa Monica, Score:1170, location: 34.01978,-118.45255, Schools: 20, Starbucks: 2, Night Clubs: 4, Train Stations: 0 Airports: 6


#### Score Montreuil

In [15]:
ix.main('126 Rue de Lagny, 93100 Montreuil')

City: Montreuil, Score:1400, location: 48.84938,2.41788, Schools: 20, Starbucks: 0, Night Clubs: 14, Train Stations: 3 Airports: 2


As we can see in the score, the best location for our gaming company is Tokyo, with a total score of 1645 points.